lib/rules/eqeqeq.js

```diff
@@ -79,11 +83,17 @@ module.exports = function(context) {
                 return;
             }
 
-            context.report(
-                node, getOperatorLocation(node),
-                "Expected '{{op}}=' and instead saw '{{op}}'.",
-                {op: node.operator}
-            );
+            context.report({
+                nodeType: node.type,
+                loc: getOperatorLocation(node),
+                message: "Expected '{{op}}=' and instead saw '{{op}}'.",
+                data: { op: node.operator },
+                fix: function(fixer) {
+                    return fixer.insertTextAfter(sourceCode.getTokenAfter(node.left).range, "=");
+
+                }
+            });
+
```

Comment:
Yeah, I'm going to adjust it so you can just do `node` and `message`, since that's the common case. Then allow everything else optionally, so most will end up being:

```js
context.report({
    node: node,
    message: "Bad stuff happened."
});
```

I think this single-argument version starts to take us back from the brink of insanity with the function signature.